From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeremy Apthorp <nornagon@nornagon.net>
Date: Tue, 2 Oct 2018 17:09:15 -0700
Subject: chore_export_environment_knodecontexttagptr.patch


diff --git a/src/env.h b/src/env.h
index 9a0f98e121b38058e61eaae8003decb7c5a75e78..a22fa78258a757cd0872bc07ffce0349233277d2 100644
--- a/src/env.h
+++ b/src/env.h
@@ -1327,7 +1327,7 @@ class Environment : public MemoryRetainer {
   uint64_t thread_id_;
   std::unordered_set<worker::Worker*> sub_worker_contexts_;
 
-  static void* const kNodeContextTagPtr;
+  NODE_EXTERN static void* const kNodeContextTagPtr;
   static int const kNodeContextTag;
 
 #if HAVE_INSPECTOR
