From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nitish Sakhawalkar <nitsakh@icloud.com>
Date: Thu, 7 Mar 2019 17:02:13 -0800
Subject: src_use_check_false_in_switch_default_case.patch

Porting https://github.com/nodejs/node/pull/26502

diff --git a/src/node_crypto.cc b/src/node_crypto.cc
index c59c6ee1aadb714d19b3794e43dfdf08d70973bc..1dbe88cbf6c4bde59da6dbaf1268a0fca58f1f1d 100644
--- a/src/node_crypto.cc
+++ b/src/node_crypto.cc
@@ -3477,7 +3477,7 @@ static ManagedEVPPKey GetPublicOrPrivateKeyFromJs(
           is_public = false;
           break;
         default:
-          UNREACHABLE("Invalid key encoding type");
+          CHECK(false);
       }
 
       if (is_public) {
